plugins/org.python.pydev.parser/src/org/python/pydev/parser/visitors/NodeUtils.java

```diff
@@ -406,10 +406,16 @@ public static String getFullRepresentationString(SimpleNode node, boolean fullOn
 
                 } else {
                     //otherwise, just add another dot and keep going.
+
+                    String representationString = getRepresentationString((SimpleNode) part, true);
+                    if (representationString.contentEquals("!<MissingName>!")) {
```

Comment:
Just to be safe, please do the comparison the other way around (as `representationString` could be `null` here).

i.e.: `"!<MissingName>!".contentEquals(representationString)`